addons/delete-others-option/userscript.js

```diff
@@ -0,0 +1,76 @@
+var addonGlobal;
+
+import {insertAfter, queryByText, removeClassContainingText, getSiblings, getMutationAddedNode} from "./utils.js";
+
+function deleteOtherCostumes(contextMenu){
+    console.log("delete others event called")
+    const currentCostumeSelector = contextMenu.parentNode.parentNode
+    const costumeSelectorsContainer = currentCostumeSelector.parentNode
+    const costumeSelectors = Array.from(costumeSelectorsContainer.children)
+    for(var i = costumeSelectors.length-1;i>=0;i--){
+        const costumeSelector = costumeSelectors[i]
+        if(costumeSelector !== currentCostumeSelector){
+            contextMenu = costumeSelector.querySelector(".react-contextmenu")
+            //const deleteBtn = queryByText(contextMenu,"delete")
+            //console.log(i, deleteBtn.outerHTML)
+            //deleteBtn.click();
+            const restoreCostumeFun = addonGlobal.tab.traps.vm.deleteCostume(i);
+            addonGlobal.tab.redux.dispatch({ 
+                type: 'scratch-gui/restore-deletion/RESTORE_UPDATE',
+                state: { restoreFun: restoreCostumeFun, deletedItem: 'Costume' }
+            });
+        }
+       
+    }
+}
+
+function addDeleteOthersOption(contextMenu){
+    const deleteNode = queryByText(contextMenu, "delete");
```

Comment:
I don't think this would work when any language other than English is used.

Reply:
That's a good point. Unfortunately delete has no id. I'll change it to place it in the 3rd position.

Reply:
unfortunately it seems delete moves about a bit so putting it in the 3rd position would some time make it out of place.

Reply:
I found out delete is always the third node with the class "react-contextmenu-item" so I can now get it without the text "delete"